kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/RealRenderContext.kt

```diff
@@ -67,15 +68,25 @@ class RealRenderContext<StateT, OutputT : Any>(
     }
   }
 
-  // @formatter:off
-  override fun <ChildInputT, ChildOutputT : Any, ChildRenderingT>
-      renderChild(
-        child: Workflow<ChildInputT, ChildOutputT, ChildRenderingT>,
-        input: ChildInputT,
-        key: String,
-        handler: (ChildOutputT) -> WorkflowAction<StateT, OutputT>
-      ): ChildRenderingT {
-    // @formatter:on
+  override fun <A : WorkflowAction<StateT, OutputT>> makeSink(): Sink<A> {
+    checkNotFrozen()
+
+    return object : Sink<A> {
+      override fun send(value: A) {
+        // If this returns false, we lost the race with another event being sent.
+        check(nextUpdateFromEvent.complete(value)) {
+          "Expected to successfully deliver $value. Are you using an old rendering?"
```

Comment:
Don't know why I didn't do this initially, but I think it would be helpful for debugging to include the deferred itself in this message, to see what we lost to.

Reply:
https://github.com/square/workflow/issues/539